libsplinter/src/admin/service/shared.rs

```diff
@@ -1017,6 +1023,94 @@ impl AdminServiceShared {
         }
     }
 
+    #[cfg(feature = "circuit-abandon")]
+    fn abandon_circuit(&mut self, circuit_id: &str) -> Result<(), ServiceError> {
+        // Verifying the circuit is able to be abandoned
+        let stored_circuit = self
+            .admin_store
+            .get_circuit(circuit_id)
+            .map_err(|err| {
+                ServiceError::UnableToHandleMessage(Box::new(AdminSharedError::SplinterStateError(
+                    format!("error occurred when trying to get circuit {}", err),
+                )))
+            })?
+            .ok_or_else(|| {
+                ServiceError::UnableToHandleMessage(Box::new(AdminSharedError::SplinterStateError(
+                    format!(
+                        "Received purged request for a circuit that does not exist: circuit id {}",
+                        circuit_id
+                    ),
+                )))
+            })?;
+
+        // send ABANDONED_CIRCUIT message to all other members' admin
+        // services
+        if let Some(ref network_sender) = self.network_sender {
+            let mut abandoned_circuit = AbandonedCircuit::new();
+            abandoned_circuit.set_circuit_id(circuit_id.to_string());
+            abandoned_circuit.set_member_node_id(self.node_id.clone());
+            let mut msg = AdminMessage::new();
+            msg.set_message_type(AdminMessage_Type::ABANDONED_CIRCUIT);
+            msg.set_abandoned_circuit(abandoned_circuit);
+
+            let envelope_bytes = msg.write_to_bytes().map_err(|err| {
+                ServiceError::UnableToHandleMessage(Box::new(MarshallingError::ProtobufError(err)))
+            })?;
+            for member in stored_circuit.members().iter() {
+                if member != &self.node_id {
+                    network_sender.send(&admin_service_id(member), &envelope_bytes)?;
+                }
+            }
+        }
+
+        if stored_circuit.circuit_status() != &StoreCircuitStatus::Active {
+            return Err(ServiceError::UnableToHandleMessage(Box::new(
+                AdminSharedError::SplinterStateError(format!(
+                    "Attempting to abandon active circuit {}",
+                    circuit_id
+                )),
+            )));
+        }
+        let (abandoned_proto_circuit, abandoned_store_circuit) = self
+            .make_abandoned_circuit(&stored_circuit)
+            .map_err(|err| ServiceError::UnableToHandleMessage(Box::new(err)))?;
+        // Updating the corresponding `active` circuit from the admin store
+        // and then removing the corresponding `CircuitProposal` from the
+        // disband request
```

Comment:
disband -> abandon

Reply:
Fixed the second half of this comment to say that it's updating the " `active` circuit in the admin store to have an `Abandoned` `circuit_status`. "